nucypher/blockchain/eth/clients.py

```diff
@@ -229,27 +230,37 @@ def sync(self,
         now = maya.now()
         start_time = now
 
+        def has_latest_block():
+            # check that our local chain data is up to date
+            return (
+                datetime.datetime.now() -
+                datetime.datetime.fromtimestamp(
+                    self.w3.eth.getBlock(self.w3.eth.blockNumber)['timestamp']
+                )
+            ).seconds < 30
```

Comment:
I think this is wrong. (1 day + 1 second).seconds = 1, and it is clearly less than 30 (but the timedelta is more than 1 day!).

Why not do `time.time() - ...['timestamp'] < 30`?
(alternatively, you can do `timedelta(seconds=30)` and compare the object with that

Reply:
Besides, I'd also check that `self.w3.eth.syncing` is False in addition to this

Reply:
that gets checked further down

Reply:
The issue this solves (assuming correct time math) is that we weren't checking for _synced_ chains, so the `while not self.syncing:` was hanging.

Reply:
Yep, right